subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/excludes/simple/DefaultExcludeAnyOf.java

```diff
@@ -44,20 +44,33 @@ protected String getDisplayName() {
 
     @Override
     public boolean excludes(ModuleIdentifier module) {
-        return components().anyMatch(e -> e.excludes(module));
+        for (ExcludeSpec component : getComponents()) {
+            if (component.excludes(module)) {
+                return true;
+            }
+        }
+        return false;
     }
 
     @Override
     public boolean excludesArtifact(ModuleIdentifier module, IvyArtifactName artifactName) {
-        return components().anyMatch(e -> e.excludesArtifact(module, artifactName));
+        for (ExcludeSpec component : getComponents()) {
+            if (component.excludesArtifact(module, artifactName)) {
+                return true;
+            }
+        }
+        return false;
     }
 
     @Override
     public boolean mayExcludeArtifacts() {
         if (mayExcludeArtifacts != null) {
             return mayExcludeArtifacts;
         }
-        mayExcludeArtifacts = components().anyMatch(ExcludeSpec::mayExcludeArtifacts);
+        mayExcludeArtifacts = false;
+        for (ExcludeSpec component : getComponents()) {
+            mayExcludeArtifacts |= component.mayExcludeArtifacts();
```

Comment:
💡 consider breaking out of the loop early once `component.mayExcludeArtifacts()` returns true